### PR TITLE
Revamp CI

### PR DESCRIPTION
**Pull request by abretaud**

### CODE REVIEW

**Comment by abretaud** [comment]
@mboudet https://github.com/genouest/biomaj-user/pull/3 doesn't seem to be applied

**Comment by mboudet** [comment]
Alright, still crashing, but at a later step.

**Comment by mboudet** [comment]
So!

* There seems to be an incompatibility between the python lib protobuf, and the system package protobuf-compiler. This can be fixed by manually setting the protobuf to '3.20.3', the max compatible version. I did it by manually installing it in the dockerfile, but it might be better to set it as a requirement in the various microservices
* OR, we could try updating the debian version. We are currently using debian:buster, we could try using debian:bullseye. I believe Olivier already created [Dockerfile-deb11](https://github.com/genouest/biomaj-docker/blob/master/Dockerfile-deb11) to that end, so we could just switch the Dockerfile with this one (after checking if everything is fine) -> I feel like it would be better in the long-term.

* The 'publish' job should be restricted to tags / pushs to the master. An example [here](https://github.com/askomics/flaskomics/blob/dev/.github/workflows/github-publish-flaskomics.yml)

@rsiminel I'll let you check?

**Comment by abretaud** [comment]
- Yep, let's pin the version of protobuf
- Yep, let's use a more recent debian version! bookworm maybe?

**Comment by mboudet** [comment]
Might not need to do both (3.20.3 is the breaking point, should be compatible after).

However, we should pin all deps anyway..

File: Dockerfile [hunk 1]
```diff
@@ -1,119 +1,102 @@
...
-RUN buildDeps='gcc python3-dev protobuf-compiler' \
+RUN buildDeps="gcc python3-dev protobuf-compiler" \
     && set -x \
+    && apt-get update \
     && apt-get install -y $buildDeps --no-install-recommends \
-    && cd /root/biomaj-core && python3 setup.py build && pip3 install . \
-    && cd /root/biomaj-zipkin && python3 setup.py build && pip3 install . \
-    && cd /root/biomaj-user && python3 setup.py build && pip3 install . \
-    && cd /root/biomaj-cli && python3 setup.py build && pip3 install . \
+    && cd /root/biomaj-core && python3 setup.py build && pip3 install --no-cache-dir . \
```
Comment (abretaud) [inline Dockerfile]: We should use this quicker way: `pip install git+https://github.com/genouest/biomaj-core.git` (for all of these modules)

Comment (rsiminel) [reply Dockerfile]: Nice.
Using this neat little trick, does pip even use the `setup.py` file from the repo that it builds its wheel from or does it look at the `requirements.txt` all by itself ?

**Comment by rsiminel** [comment]
I just created a PR, here is what it does and does not do:

> Yep, let's pin the version of protobuf
- Looking at the `requirements.txt` of each repo, it seems that only [`biomaj-process`](https://github.com/genouest/biomaj-process/tree/fix_protobuf2) and [`biomaj-download`](https://github.com/genouest/biomaj-download/tree/fix_protobuf) require protobuf and conveniently, @mboudet has an open branch to fix protobuf in each of those repos. Shall we pin the version there ?
> Yep, let's use a more recent debian version!
- Done using bullseye. Looking through the file history, it seems that Oliver's [Dockerfile-deb11](https://github.com/genouest/biomaj-docker/blob/master/Dockerfile-deb11) was just a copy-paste of the Dockerfile as it was two years ago with `FROM debian:buster` changed to `FROM debian:bullseye`. Ok to remove now ?
> The 'publish' job should be restricted to tags / pushs to the master.
- Done. I just added an `if: github.ref == 'refs/heads/master'` to the publish job.
> We should use this quicker way: pip install git+https://github.com/genouest/biomaj-core.git (for all of these modules)
- Done but I'm surprised at how easy that was, can you guys make sure I didn't get rid of anything important ?

File: Dockerfile (hunk 1 above)
Comment (abretaud) [reply Dockerfile]: it uses setup.py

**Comment by rsiminel** [comment]
So now this PR closes #5 
Thanks for making it possible to get rid of those pesky `git clone`s mboudet!

**Comment by mboudet** [comment]
Added changes:

* Add `#!/bin/env python3` to the various CLI files in the modules

The CLI files are properly installed with the pip install, just not in /root/*module*/cli.
`#!/bin/env python3` is required since we can't directly call 'python3' on a file in the PATH. (We could try to use an absolute path to the file, but not sure if that's reliable).
Another issue is that `#!/bin/env python3` works as expected, but `#!/bin/env python` uses the conda python instead, so there is a module not found error.

Not really fond of this solution to be honest, but it works... for now.

* Updated prometheus

Not much of a choice here. I tried rebuilding the 1.8.2 image, but it doesn't work anymore. At most, people will lose the last 720 hours data, which is not that much I feel like. Not sure if anyone actually use the prometheus to be honest.
There might be a 'proper' upgrade procedure, but it's out of the scope of this repo.

File: CHANGES.md [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 Notes: requires mongo 3.x
 
-## shahikorma
+## v14 - TBR
+
+/!\ Breaking: Upgrading promoetheus version from 2.45.6. Data will be lost in the upgrade.
```
Comment (abretaud) [inline CHANGES.md]:
```suggestion
/!\ Breaking: Upgrading prometheus version from 2.45.6. Data will be lost in the upgrade.
```

File: Dockerfile [hunk 3]
```diff
@@ -1,119 +1,86 @@
...
-    && add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable" \
     && apt-get update \
-    && apt-get install -y docker-ce-cli \
-    && apt-get purge -y --auto-remove $buildDeps
+    && apt-get install --no-install-recommends -y docker-ce-cli \
+    && apt-get purge -y --auto-remove $buildDeps \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
 
 ENV BIOMAJ_RELEASE="shahikorma-v13"
```
Comment (abretaud) [inline Dockerfile]:
```suggestion
ENV BIOMAJ_RELEASE="14"
```

**Review by mboudet — APPROVED** [review]

**Pull request merged**